src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -667,29 +667,46 @@ func (c controller) UpdateBMHs(ctx context.Context, wg *sync.WaitGroup) {
 		c.log.Infof("Finished UpdateBMHs")
 		wg.Done()
 	}()
+
+	var cluster *models.Cluster
+	var err error
```

Comment:
We get the cluster in the `SetReadyState` you can get it from there and just not start this goroutine.

Reply:
Optional yes